lib/plugins/stonith/external/drac5.in

```diff
@@ -6,19 +6,45 @@
 # License:      GNU General Public License (GPL)
 #
 
-trap 'if [ -n "$outf" ]; then ha_log.sh err "`cat $outf`"; rm -f "$outf"; fi' 0
-outf=`mktemp` || {
-	ha_log.sh err "mktemp failed"
-	exit 1
+ha_log() {
+	severity=$1
+	outfile=$2
+	if [ -n "$outfile" ]
+	then
+		if [ -s "$outfile" ]
+		then
+			ha_log.sh "$severity" "`cat $outfile`"
+		fi
+	fi
+}
+
+cleanup () {
+	if [ -e "$outf" ]
+	then
+		rm -r "$outf"
+	fi
 }
 
 sshlogin() {
+	outf=`mktemp` || {
+		ha_log.sh err "mktemp failed"
+		return 1
+	}
+
 	if [ x = "x$ipaddr" -o x = "x$userid" ]
 	then
 		ha_log.sh err "ipaddr or userid missing; check configuration"
 		return 1
 	fi
 	@SSH@ -q -x -n $userid@$ipaddr racadm serveraction "$1" >$outf 2>&1
+	if [ $? -ne 0 ]
+	then
+		ha_log err "$outf"
```

Comment:
The function should return the racadm's exit code, i.e. if it reports an error then the action failed (I'd assume). Note that the following call to cleanup will effectively clear the exit code, so you have to save it.